### DISCUSSION TITLE
Language interop with rust?

### DISCUSSION THREAD

**Discussion by ezntek** [nushell/nushell#9302] [Q&A]
I've been using nu for a few months now, and I'm just wondering, is there language interop support between nu and rust? (Just like C and Python with ffi et al., Python and Rust with pyo3 and all that fun stuff)

**Comment by amtoine** [comment]
not that i'm aware of :open_mouth:

**Reply by amtoine** [reply]
this would be for instance to call Nushell commands from Rust?

**Reply by ezntek** [reply]
yes

**Reply by amtoine** [reply]
can't we run any shell from Rust code?
something like `exec("nu", "ls", "-l")` :thinking:

**Comment by RayyanNafees** [comment]
there are nushell plugins that can be made using rust, python, javascript now !
